pkg/controllers/providers/instancetype/memoryoverhead/controller.go

```diff
@@ -0,0 +1,69 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package memoryoverhead
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/karpenter-provider-aws/pkg/providers/instancetype"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+	"time"
+)
+
+type Controller struct {
+	kubeClient           client.Client
+	instancetypeProvider instancetype.Provider
+}
+
+func NewController(kubeClient client.Client, instancetypeProvider instancetype.Provider) *Controller {
+	return &Controller{
+		kubeClient:           kubeClient,
+		instancetypeProvider: instancetypeProvider,
+	}
+}
+
+func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	ctx = injection.WithControllerName(ctx, "instancetypes.memoryoverhead")
+	if err := c.instancetypeProvider.UpdateInstanceTypeMemoryOverhead(ctx, c.kubeClient); err != nil {
+		return reconcile.Result{}, fmt.Errorf("updating instancetype, %w", err)
+	}
+	return reconcile.Result{RequeueAfter: 12 * time.Hour}, nil
+}
+
+func (c *Controller) Register(_ context.Context, m manager.Manager) error {
+	return controllerruntime.NewControllerManagedBy(m).
+		Named("instancetypes.memoryoverhead").
+		For(&corev1.Node{}).
+		WithEventFilter(predicate.Funcs{
+			// Only trigger reconciliation on node creation events
+			CreateFunc: func(_ event.CreateEvent) bool {
+				return true
```

Comment:
If we're making it a precondition for being a karpenter node, we'd probably need to change this to update, rather than create, since we first create the node, but then update it with the correct labels

Reply:
Thanks, good call! Fixed